src/odb/src/db/dbTrackGrid.cpp

```diff
@@ -271,4 +312,24 @@ void dbTrackGrid::destroy(dbTrackGrid* grid_)
   block->_track_grid_tbl->destroy(grid);
 }
 
+// User Code Begin PrivateMethods
+void _dbTrackGrid::getAverageTrackPattern(bool is_x,
+                                          int& track_init,
+                                          int& num_tracks,
+                                          int& track_step)
+{
+  std::vector<int> coordinates;
+  dbTrackGrid* track_grid = (dbTrackGrid*) this;
+  if (is_x) {
+    track_grid->getGridX(coordinates);
```

Comment:
I am not sure if this is the proper way to call a public method in the private context. I am accessing the "this" object without the underline. Is this allowed?

Reply:
That's fine.